core/db/queries.py

```diff
@@ -26,13 +26,43 @@ def filter_on_regions(itl_regions: set[str]) -> list[UUID]:
     return updated_results
 
 
+def query_extend_with_outcome_filter(base_query, outcome_categories: list[str] | None = None):
+    """
+    Extend base query to select additional columns for OutcomeDim / OutcomeData.
```

Comment:
I'm not sure this description is totally accurate, can it say something more like: 

"""
Extend base query to include join to OutcomeDim / OutcomeData.

Conditionally apply a filter on OutcomeDim catergory field
...
"""